common/domain/dlq_message_handler_test.go

```diff
@@ -359,3 +502,49 @@ func (s *dlqMessageHandlerSuite) TestMergeMessages_IgnoreErrorOnUpdateDLQAckLeve
 	s.NoError(err)
 	s.Nil(token)
 }
+
+func (s *dlqMessageHandlerSuite) TestMergeMessages_NonDomainTask() {
+	ackLevel := int64(10)
+	lastMessageID := int64(20)
+	pageSize := 100
+	pageToken := []byte{}
+
+	// Create a task that mimics a non-domain replication task by not setting DomainTaskAttributes
+	tasks := []*types.ReplicationTask{
+		{
+			TaskType:     types.ReplicationTaskTypeDomain.Ptr(), // Still set to domain but no attributes
+			SourceTaskID: 1,
+		},
+	}
+
+	s.mockReplicationQueue.EXPECT().GetDLQAckLevel(gomock.Any()).Return(ackLevel, nil).Times(1)
+	s.mockReplicationQueue.EXPECT().GetMessagesFromDLQ(gomock.Any(), ackLevel, lastMessageID, pageSize, pageToken).
+		Return(tasks, nil, nil).Times(1)
+
+	token, err := s.dlqMessageHandler.Merge(context.Background(), lastMessageID, pageSize, pageToken)
+
+	s.NotNil(err)
+	s.IsType(&types.InternalServiceError{}, err)
+	s.Equal("Encounter non domain replication task in domain replication queue.", err.Error())
+	s.Nil(token)
+}
+
+func (s *dlqMessageHandlerSuite) TestEmitDLQSizeMetricsLoop_ErrorHandling() {
+	expectedError := fmt.Errorf("error fetching DLQ size")
+	s.mockReplicationQueue.EXPECT().GetDLQSize(gomock.Any()).Return(int64(0), expectedError).AnyTimes()
+
+	// Start the metrics loop in a goroutine
+	go s.dlqMessageHandler.emitDLQSizeMetricsLoop()
+
+	// Allow some time for the goroutine to run and tick at least once
+	time.Sleep(1 * time.Nanosecond)
+
+	// Close the done channel to signal the loop to stop
+	close(s.dlqMessageHandler.done)
+
+	// Wait a bit to ensure the loop exits
+	time.Sleep(1 * time.Nanosecond)
+
+	s.dlqMessageHandler.logger.Warn("Failed to get DLQ size.", tag.Error(errors.New("DomainReplicationQueueSizeLimit")))
```

Comment:
did you mean to add an expected warning call here instead of actually calling warn function?

Reply:
yes. you are just adding another log. no validation here. 
as mentioned in other comment let's ignore this edge case. not much value in validating warning log is there